docs/contributing-to-analyses/analysis-modules/script-structure.md

```diff
@@ -0,0 +1,124 @@
+```
+
+#### Parse input arguments { #r-arguments }
+
+We recommend using the [`optparse` package](https://cran.r-project.org/web/packages/optparse/index.html) to parse input arguments in R scripts.
+
+For example, after loading the `optparse` package, you can define input options with the [`make_option()`](https://rdrr.io/cran/optparse/man/add_make_option.html) function, and then use the [`parse_args()`](https://rdrr.io/cran/optparse/man/parse_args.html) function to parse them into a usable list.
+
+```r
+# Set up arguments
+option_list <- list(
+  make_option(
+    # Users can specify this argument with either `-i` or `--input_rds_file`
+    opt_str = c("-i", "--input_rds_file"),
+    # The provided value will be stored as a character variable called `input_anndata_file`
+    type = "character",
+    dest = "input_rds_file",
+    # Description of the argument
+    help = "path to input RDS file for analysis"
+  ),
+  ##  define additional options here ##
+)
+
+# Parse arguments into a usable list
+opt_list <- parse_args(OptionParser(option_list = option_list))
+```
+
+In the above example, you can access the parsed argument as `opt_list$input_rds_file`.
+
+#### Additional R tips
+
+- You can add more structure to your script using [sectioning comments](https://r4ds.hadley.nz/workflow-style#sectioning-comments), which help you navigate the code in RStudio much like section headers in a notebook.
+
+### Considerations for Python scripts
+
+#### Shebang { #python-shebang }
+
+Your Python scripts should always start with this line:
+
+```python
+#!/usr/bin/env python3
+```
+
+#### Parse input arguments { #python-arguments }
+
+We recommend using the [`argparse` package](https://docs.python.org/3/library/argparse.html) to parse input arguments in Python scripts.
+
+For example, after loading the `argparse` package, you can define a parser object with [`argparse.ArgumentParser()`](https://docs.python.org/3/library/argparse.html#creating-a-parser), and then use [`.add_argument()`](https://docs.python.org/3/library/argparse.html#adding-arguments) to add each argument to the parser object.
+
+
+```python
+# define a parser object with helpful information for script users
+parser = argparse.ArgumentParser(
+    description = "Performs task XYZ as part of analysis"
+)
+
+# parse a command line argument
+parser.add_argument(
+    # Users can specify this argument with either `-i` or `--input_anndata_file`
+    "-i",
+    "--input_anndata_file",
+    # The provided value will be stored as a string called `input_anndata_file`
+    type = "character",
+    dest = "input_adddata_file",
+    # Description of the argument
+    help = "path to input AnnData file for analysis"
+)
+```
+
+In the above example, you can access the parsed argument as `parser.input_anndata_file`.
+
+
+#### Additional Python tips
+
+- You may wish to enclose your opening comments inside a [triple-quote block](https://www.geeksforgeeks.org/triple-quotes-in-python/).
```

Comment:
I don't know that this is worth including? (Also, I am inordinately annoyed by the curly quotes in their syntax block) But I might include something about docstrings specifically: https://peps.python.org/pep-0257/ or https://www.geeksforgeeks.org/python-docstrings/